pyquil/reference_simulator.py

```diff
@@ -163,8 +177,42 @@ class ReferenceDensitySimulator(AbstractQuantumSimulator):
     def __init__(self, n_qubits: int, rs: RandomState = None):
         self.n_qubits = n_qubits
         self.rs = rs
-        self.density = np.zeros((2 ** n_qubits, 2 ** n_qubits), dtype=np.complex128)
-        self.density[0, 0] = complex(1.0, 0)
+        self.initial_density = np.zeros((2 ** n_qubits, 2 ** n_qubits), dtype=np.complex128)
```

Comment:
This could all be replaced with `self.set_initial_state(None)`.

Reply:
- [x] todo